userspace/libsinsp/k8s_pod_handler.cpp

```diff
@@ -86,11 +86,14 @@ k8s_pod_handler::k8s_pod_handler(k8s_state_t& state
 	,bt_ptr_t bt
 	,bool connect
 	,bool blocking_socket
+	,std::string node_selector
 #endif // HAS_CAPTURE
 	):
 		k8s_handler("k8s_pod_handler", true,
 #if defined(HAS_CAPTURE) && !defined(_WIN32)
-					url, "/api/v1/pods?fieldSelector=status.phase!=Failed,status.phase!=Unknown,status.phase!=Succeeded",
+					url, 
+					"/api/v1/pods?fieldSelector=status.phase!=Failed,status.phase!=Unknown,status.phase!=Succeeded"
+					+ (node_selector.size() == 0 ? "" : ",spec.nodeName=" + node_selector),
```

Comment:
This isn't exactly pretty but we can't make it much better in an initializer expression I think

Reply:
I thought the same thing, just opted for the less invasive approach. :smile_cat: